core/app/components/FormBuilder/actions.ts

```diff
@@ -42,18 +66,32 @@ export const saveForm = defineServerAction(async function saveForm(form: FormBui
 	const { elements, formId, access } = form;
 	//todo: this logic determines what, if any updates to make. that should be determined on the
 	//frontend so we can disable the save button if there are none
-	const { upserts, deletes } = elements.reduce<{
+	const { upserts, deletes, relatedPubTypes, deletedRelatedPubTypes } = elements.reduce<{
 		upserts: NewFormElements[];
 		deletes: FormElementsId[];
+		relatedPubTypes: NewFormElementToPubType[];
+		deletedRelatedPubTypes: FormElementsId[];
 	}>(
 		(acc, element, index) => {
 			if (element.deleted) {
 				if (element.elementId) {
 					acc.deletes.push(element.elementId);
 				}
 			} else if (!element.elementId) {
-				// Newly created elements have no elementId
-				acc.upserts.push(formElementsInitializerSchema.parse({ formId, ...element }));
+				// Newly created elements have no elementId, so generate an id to use
+				const id = crypto.randomUUID() as FormElementsId;
```

Comment:
note: now setting the id here instead of letting the db to do it so that we can have access to the id when creating the pub type relationship too